upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -56,7 +80,83 @@ func (_ *AttachISO) CheckChanges(a, e, changes *AttachISO) error {
 	return nil
 }
 
-func (_ *AttachISO) RenderVC(t *vsphere.VSphereAPITarget, a, e, changes *AttachISO) error {
-	glog.Info("AttachISO.RenderVC invoked!")
+func (_ *AttachISO) RenderVSphere(t *vsphere.VSphereAPITarget, a, e, changes *AttachISO) error {
+	startupScript, err := changes.BootstrapScript.ResourceNodeUp(changes.IG)
+	startupStr, err := startupScript.AsString()
+	if err != nil {
+		return fmt.Errorf("error rendering startup script: %v", err)
+	}
+	dir, err := ioutil.TempDir("", *changes.VM.Name)
+	defer os.RemoveAll(dir)
+
+	isoFile := createISO(changes, startupStr, dir)
+	err = t.Cloud.UploadAndAttachISO(changes.VM.Name, isoFile)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func createUserData(startupStr string, dir string) {
+	// Update the startup script to add the extra spaces for
+	// indentation when copied to the user-data file.
+	strArray := strings.Split(startupStr, "\n")
+	for i, str := range strArray {
+		if len(str) > 0 {
+			strArray[i] = "       " + str
+		}
+	}
+	startupStr = strings.Join(strArray, "\n")
+
+	data := strings.Replace(userDataTemplate, "$SCRIPT", startupStr, -1)
+	userDataFile := filepath.Join(dir, "user-data")
+	glog.V(4).Infof("User data file content: %s", data)
+
+	if err := ioutil.WriteFile(userDataFile, []byte(data), 0644); err != nil {
+		glog.Fatalf("Unable to write user-data into file %s", userDataFile)
+	}
+	return
+}
+
+func createMetaData(dir string, vmName string) {
+	data := strings.Replace(metaDataTemplate, "$INSTANCE_ID", uuid.NewUUID().String(), -1)
+	data = strings.Replace(data, "$LOCAL_HOST_NAME", vmName, -1)
+
+	glog.V(4).Infof("Meta data file content: %s", string(data))
+
+	metaDataFile := filepath.Join(dir, "meta-data")
+	if err := ioutil.WriteFile(metaDataFile, []byte(data), 0644); err != nil {
+		glog.Fatalf("Unable to write meta-data into file %s", metaDataFile)
+	}
+	return
+}
+
+func createISO(changes *AttachISO, startupStr string, dir string) string {
+	createUserData(startupStr, dir)
+	createMetaData(dir, *changes.VM.Name)
+
+	isoFile := filepath.Join(dir, *changes.VM.Name+".iso")
+	var cmd *exec.Cmd
+
+	switch os := runtime.GOOS; os {
+	case "darwin":
+		cmd = exec.Command("mkisofs", "-o", isoFile, "-volid", "cidata", "-joliet", "-rock", dir)
+	case "linux":
+		cmd = exec.Command("genisoimage", "-o", isoFile, "-volid", "cidata", "-joliet", "-rock", dir)
+	default:
+		glog.Fatalf("Cannot generate ISO file %s. Unsupported operation system!!!", isoFile)
+	}
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		glog.Fatalf("Error %s occurred while executing command %+v", err, cmd)
+	}
+	glog.V(4).Infof("genisoimage std output : %s\n", out.String())
+	glog.V(4).Infof("genisoimage std error : %s\n", stderr.String())
+	return isoFile
```

Comment:
Please save command name in a variable and use that for logging. It could be genisoimage or mkisofs.